slice.go

```diff
@@ -592,3 +592,12 @@ func IsSortedByKey[T any, K constraints.Ordered](collection []T, iteratee func(i
 
 	return true
 }
+
```

Comment:
This function makes sense in this file to me since it's only really useful for slices. Happy to move it though.

Also happy to change the name.